### PR TITLE
Feature/basicwlfinalization

### PR DESCRIPTION
**Pull request by NikolaiMe**
Added BasicWL support plus tests

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -58,6 +58,7 @@
         "tsup": "^8.1.0",
         "typescript": "^5.5.3",
         "typescript-eslint": "^8.15.0",
-        "xmllint-wasm": "^4.0.2"
+        "xmllint-wasm": "^4.0.2",
+        "zod-to-ts": "^1.2.0"
```
Comment (schwarmco) [inline package.json:62]: is `zod-to-ts` already in use afar from the playground-test?

File: src/profiles/convert.ts [hunk 2]
```diff
@@ -2,64 +2,84 @@
...
 import { TextTypeConverter } from '../types/udt/TextTypeConverter.js'
 
-type ArrayDotNotation<T, Prefix extends string> = T extends (infer U)[]
-    ? `${Prefix}.${number}` | (U extends object ? DotNotation<U, `${Prefix}.${number}.`> : never)
+// Main DotNotation type that delegates to ArrayDotNotation for array
+export type DotNotation<T> = T extends object
+    ? {
+          [K in keyof T & string]:
+              | K
+              | (NonNullable<T[K]> extends (infer U)[]
```
Comment (schwarmco) [inline src/profiles/convert.ts:30]: that's clever!

File: src/profiles/convert.ts [hunk 3]
```diff
@@ -84,7 +104,7 @@ export abstract class Converter<Profile, ProfileXml> {
                 continue
             }
 
-            if (item.arrayMap && Array.isArray(value)) {
+            if (Array.isArray(value) && !(item.converter instanceof SpecifiedTaxRegistrationsForSellerTypeConverter)) {
```
Comment (schwarmco) [inline src/profiles/convert.ts:107]: that's a very specific edge case check in the converter here - is this guaranteed to only be the case for the sellers tax numbers or could this affect other converters too (i.e. in extended profile)? 

if so, what do you think about having a flag in the converter that specifies this, i.e.:

```ts
if (Array.isArray(value) && item.converter.isArrayable) {
```

File: package.json (hunk 1 above)
Comment (NikolaiMe) [reply package.json:62]: correct I use it to create and print-out the full ts-type. It helps a lot when the XSD validation fails and I need to find out what's the bug in our profile. And I do this only via playground tests. But I don't use it somewhere else.
Btw. I basically only have this playground test because 'npx ts-node whateverfile.ts' is not working in our setup. It always tells me that .ts is a unknown file-format, which is quite weird because it's **ts**-node... And in other projects this works perfectly. Do you have an idea why this is not working? I already tried to fix it by adapting the ts-config, but this messed up everything. So I had this workaround to execute the code via a test...

File: src/profiles/convert.ts (hunk 2 above)
Comment (NikolaiMe) [reply src/profiles/convert.ts:30]: Yeah more or less clever :-D ... this is the result of one hour researching, trial and error and useless Github Co-Pilot discussions. But finally it worked and didn't mess up the optional keys anymore.

File: src/profiles/convert.ts (hunk 3 above)
Comment (NikolaiMe) [reply src/profiles/convert.ts:107]: Yeah this was a quick and dirty solution for the problem, that this single converter takes an array and maps the items of that array to two keys in an object.
I didn't check whether this might be an issue in the future, too tbh....

Wait a second I just found a bug. Because when talking about that "single converter that takes an array" I remembered that TradeAllowanceChargeTypeConverter also takes an Array, and I didn't handle that yet, and the tests do not check this yet, as the test-pdf does not have allowances or charges.

--> Long story short: We should use your implementation because we already have two of those edge-cases already.

Plus: I don't like it, that the arrays in "our-schemes" are unions right now (either object or array). This makes the handling of the data unnecessarily hard (because you always need to check, is it an array or is it an object because you can use your data). I would prefer to always have an array instead... And I think we could also use your solution to make this happen...

I try to implement this.

Comment (NikolaiMe) [reply src/profiles/convert.ts]: I implemented it a little bit differently. Because my solution solves both issues (specific excludes in convert.ts and unions object | object[] removed from object scheme). Plus I even more embrace the Converter logic, which I think is more consistent than the old implementation.

See details below

File: package.json (hunk 1 above)
Comment (NikolaiMe) [reply package.json:62]: Do you want me to take any actions for this? Shall I remove the dependency (i would rather keep it, because I will need it in the future again)

Comment (schwarmco) [reply package.json:62]: no, thats fine if we're going to use it - i will look into the other issue with ts-node you mentioned though
